packages/arb-token-bridge-ui/src/hooks/useNetworksAndSigners.tsx

```diff
@@ -156,18 +156,38 @@ export function NetworksAndSignersProvider(
     async (web3Provider: Web3Provider, address: string) => {
       const providerChainId = (await web3Provider.getNetwork()).chainId
 
+      // If provider is not supported, display warning message
+      if (!(providerChainId in chainIdToDefaultL2ChainId)) {
+        console.error(`Provider chainId not supported: ${providerChainId}`)
+        setResult({ status: UseNetworksAndSignersStatus.NOT_SUPPORTED })
+      }
+
+      /***
+       * Case 1: selectedChainId is undefined => set it to provider's default L2
+       * Case 2: selectedChainId is defined but not supported by provider => reset query params -> case 1
+       * Case 3: selectedChainId is defined and supported, continue
+       */
       let _selectedL2ChainId = selectedL2ChainId
+      const providerSupportedL2 = chainIdToDefaultL2ChainId[providerChainId]
+
+      // Case 1: use a default L2 based on the connected provider chainid
+      _selectedL2ChainId = _selectedL2ChainId || providerSupportedL2[0]
       if (_selectedL2ChainId === undefined) {
```

Comment:
This check should be:

```suggestion
if (typeof _selectedL2ChainId === 'undefined') {
```

But, this should never be the case, right? It's mostly to make TypeScript happy?

Reply:
Right now, it's to be safe, this would only happen if a L1 had no corresponding L2 in `chainIdToDefaultL2ChainId`